File: src/com/jupiter/ganymede/math/function/LinearFunction.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.jupiter.ganymede.math.function;

/**
 *
 * @author Nathan Templon
 */
public class LinearFunction implements SingleVariableRealFunction {
    
    // Fields
    private final double slope;
    private final double intercept;
    
    
    // Properties
    public final double getSlope() {
        return this.slope;
    }
    
    public final double getIntercept() {
        return this.intercept;
    }
    
    
    // Initialization
    public LinearFunction(double slope, double intercept) {
        this.slope = slope;
        this.intercept = intercept;
    }
    
    
    @Override
    public Double apply(Double x) {
        return intercept + slope * x;
    }
    
}

File: src/com/jupiter/ganymede/math/function/LinearFunction.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.jupiter.ganymede.math.function

/**

 * @author Nathan Templon
 */
public class LinearFunction(public val slope: Double, public val intercept: Double) : SingleVariableRealFunction {

    override fun apply(x: Double?): Double? {
        return intercept + slope * x!!
    }

}
